prow/plank/reconciler.go

```diff
@@ -294,19 +299,28 @@ func (r *reconciler) defaultReconcile(ctx context.Context, request reconcile.Req
 	return *res, err
 }
 
-// serializeIfNeeded serializes the reconciliation of Jobs that have a MaxConcurrency setting, otherwise
-// multiple reconciliations of the same job may race and not properly respect that setting.
+// serializeIfNeeded serializes the reconciliation of Jobs that have a MaxConcurrency or a JobQueueName set, otherwise
+// multiple reconciliations of the same job or queue may race and not properly respect that setting.
 func (r *reconciler) serializeIfNeeded(ctx context.Context, pj *prowv1.ProwJob) (*reconcile.Result, error) {
-	if pj.Spec.MaxConcurrency == 0 {
-		return r.reconcile(ctx, pj)
+	if pj.Spec.MaxConcurrency > 0 {
```

Comment:
Instead of having almost identical blocks here, is it possible to abstract this away into a function and just call it twice with different parameters? Or is that too much overhead?

Reply:
Definitely possible, but I think in this case it would be heavily parameterized and won't improve readability.